lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -3711,23 +3720,45 @@ void IRGenDebugInfoImpl::emitTypeMetadata(IRGenFunction &IGF,
   if (!DS || DS->getInlinedFunction()->isTransparent())
     return;
 
-  llvm::SmallString<8> Buf;
-  static const char *Tau = SWIFT_UTF8("\u03C4");
-  llvm::raw_svector_ostream OS(Buf);
-  OS << '$' << Tau << '_' << Depth << '_' << Index;
-  uint64_t PtrWidthInBits = CI.getTargetInfo().getPointerWidth(clang::LangAS::Default);
+  auto id = IGF.getSwiftModule()->getASTContext().getIdentifier(Name);
+  uint64_t PtrWidthInBits =
+      CI.getTargetInfo().getPointerWidth(clang::LangAS::Default);
   assert(PtrWidthInBits % 8 == 0);
   auto DbgTy = DebugTypeInfo::getTypeMetadata(
-      getMetadataType(Name)->getDeclaredInterfaceType().getPointer(),
+      getMetadataType(Identifier)->getDeclaredInterfaceType().getPointer(),
       Size(PtrWidthInBits / 8),
       Alignment(CI.getTargetInfo().getPointerAlign(clang::LangAS::Default)));
-  emitVariableDeclaration(IGF.Builder, Metadata, DbgTy, IGF.getDebugScope(),
-                          {}, {OS.str().str(), 0, false},
-                          // swift.type is already a pointer type,
-                          // having a shadow copy doesn't add another
-                          // layer of indirection.
-                          IGF.isAsync() ? CoroDirectValue : DirectValue,
-                          ArtificialValue);
+  emitVariableDeclaration(
+      IGF.Builder, Metadata, DbgTy, IGF.getDebugScope(), {},
+      {id.str(), 0, false}, // swift.type is already a pointer type,
+                            // having a shadow copy doesn't add another
+                            // layer of indirection.
+      IGF.isAsync() ? CoroDirectValue : DirectValue, ArtificialValue);
+}
+
+void IRGenDebugInfoImpl::emitTypeMetadata(IRGenFunction &IGF,
+                                          llvm::Value *Metadata, unsigned Depth,
+                                          unsigned Index, StringRef Name) {
+  llvm::SmallString<8> Buf;
+  static const char *Tau = SWIFT_UTF8("\u03C4");
+  llvm::raw_svector_ostream OS(Buf);
+  OS << '$' << Tau << '_' << Depth << '_' << Index;
+
+  emitArtificialVariable(IGF, Metadata, OS.str(), Name);
+}
+
+void IRGenDebugInfoImpl::emitWitnessTable(IRGenFunction &IGF,
+                                          llvm::Value *Metadata, unsigned Depth,
+                                          unsigned Index,
```

Comment:
Keying this off of depth/index is insufficient. Eg in the following, we pass a witness table for the `T.Element: Equatable` conformance:

```
func f<T: Sequence>(_: T) where T.Element: Sequence {}
```